server/matchmaker.go

```diff
@@ -161,18 +161,20 @@ func NewLocalMatchmaker(logger, startupLogger *zap.Logger, config Config, router
 		activeIndexes:  make(map[string]*MatchmakerIndex),
 	}
 
-	go func() {
```

Comment:
`config.GetMatchmaker().IntervalSec` values <= 0 are not valid, so I don't think we should add this workaround. For test purposes I like calling the `processLabelUpdate` function directly in the test case, but leave the goroutine in place and just set the config `IntervalSec` to a very high value in the test cases.

Reply:
Yes I agree, that is a good suggestion.  It returns the main code to the original implementation, and now all changes are confined to the tests.

Reply:
The change has been made in the latest commit.